### PR TITLE
Simplification of notification settings and merge task settings

### PR DESCRIPTION
**Pull request by elioschmutz**
Jira: https://4teamwork.atlassian.net/browse/GEVER-27

This PR refactors and simplifies the notification settings and merges task-settings into setting-groups.

The notification-settings form already provided aliases. It was possible to define some activities and alias them to one setting. We where tracking a setting entry for each aliased item and updated each entry if the user did some modifications on his settings.

This seems to be a workaround for notification-groups. The workaround seemed to be required through historical implementation details. The settings where not really separated form activities itself. So each setting had a related activity. Through this implementation we got several problems. The naming seems to be very confused, we have strong relations from activities and settings and we're not able to group settings properly. 

Because of this facts, I decided to refactor the notification settings form:
- I tried to rename some functions and added some comments to make it easier to understand and read
- I decoupled the notification-settings handling. Now we have a clear interface to manage the notification settings
- I introduced a general notification configuration to make it much easier to extend the form with new settings

Further I implemented the following:
- I removed the notification icons (approved: https://4teamwork.atlassian.net/browse/GEVER-27?focusedCommentId=11395)
- the settings are now decoupled from activities. Each setting can relate to multiple activites
-  added a migration upgradestep to merge the old notification settings into the defined notification groups

## Before
Bildschirmfoto 2020-05-04 um 11 52 30

## After
Bildschirmfoto 2020-05-04 um 11 52 40

## Checklist (Must have)

- [x] Changelog entry
- [x] Documentation updated (notably for API and deployment)
- [x] Link to issue (Jira or GitHub) and backlink in issue (Jira)

## Checklist (optional)

- Upgrade steps (changes in profile):
  - [x] Make it deferrable if possible
  - [x] Execute as much as possible conditionally
- New translations
  - [x] All msg-strings are unicode
  - [x] Correct i18n-domain was used (Copy-Paste errors are common here)

### CODE REVIEW

File: opengever/activity/notification_settings.py [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+    },
+    {
+        'id': 'todo-modified',
+        'activities': ['todo-modified'],
+    },
+]
+
+
+class NotificationSettings(object):
+    """Object which porivdes functions to lookup the notification settings.
```
Comment (lukasgraf) [inline opengever/activity/notification_settings.py:238]: `porivdes` -> `provides`

File: opengever/activity/notification_settings.py [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+
+    def get_setting(self, setting_kind, userid):
+        """Returns a specific setting for a specific userid
+        """
+        return self.get_settings(userid).get(setting_kind)
+
+    def get_setting_by_activity_kind(self, activity_kind, userid):
+        """Each notification setting is responsible for one or multiple activities.
+
+        This function returns the setting for a speific userid which is responsible
```
Comment (lukasgraf) [inline opengever/activity/notification_settings.py:263]: speific -> specific

File: opengever/activity/browser/settings.py [hunk 3]
```diff
@@ -160,6 +158,9 @@ class NotificationSettings(BrowserView):
     The notification settings provides notification settings for specific
     activities for the current user.
     """
+    def __init__(self, context, request):
+        super(NotificationSettings, self).__init__(context, request)
```
Comment (lukasgraf) [inline opengever/activity/browser/settings.py:130]: Can we rename the `NotificationSettings` BrowserView so it doesn't clash with the other `NotificationSettings` class?

File: opengever/activity/browser/settings.py [hunk 4]
```diff
@@ -179,46 +179,49 @@ def reset_user_setting(self):
...
-                                   'setting_type': setting_type
-                                   })
+            user_settings.append({'id': setting.get('id'),
+                                  'title': title,
+                                  'help_text': help_text,
+                                  'value': value,
+                                  'setting_type': setting_type
+                                  })
 
+        # The mapping here is weired because we refactored the python part but
```
Comment (lukasgraf) [inline opengever/activity/browser/settings.py:232]: `weired` -> `weird`

File: opengever/activity/browser/settings.py [hunk 5]
```diff
@@ -179,46 +179,49 @@ def reset_user_setting(self):
...
+        # not the Handlebars (JS) part. Because we would have to update a lot
+        # of frontend-code, wel let it how it is.
```
Comment (lukasgraf) [inline opengever/activity/browser/settings.py:234]: -> `we'll leave it as is`

File: opengever/activity/notification_settings.py [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+    def _get_default_notification_settings(self):
+        """Returns the default notificiation settings.
+        """
+        if not hasattr(self, '_default_notification_settings'):
+            setattr(self, '_default_notification_settings', {
+                default.kind: default for default
+                in model.NotificationDefault.query
+                })
+
+        return getattr(self, '_default_notification_settings')
```
Comment (lukasgraf) [inline opengever/activity/notification_settings.py:323]: No need for the `getattr` here? Just `return self. _default_notification_settings ` 😉
_(Unless you wanted to factor out `_default_notification_settings ` into a constant)._

File: opengever/activity/browser/settings.py (hunk 3 above)
Comment (elioschmutz) [reply opengever/activity/browser/settings.py:130]: What do you think about `NotificationSettingsFormAPI` ?

Comment (lukasgraf) [reply opengever/activity/browser/settings.py:130]: Or just `NotificationSettingsView`?

**Pull request merged**